server/src/stateManagement.mjs

```diff
@@ -84,10 +87,32 @@ function addDefaultUser(userId) {
   state[''+userId].isDefaultUserState = true;
 }
 
+// return state based on hierarchy (From lowest priority to highest) global->group->user
 function getState(userId) {
+
   if ( userId in state ) {
+    const stateCopy = JSON.parse( JSON.stringify(state) )
+    let finalState = stateCopy['global'];
+    userId = '' + userId;
+    if( userId.includes("~")){
+      let group = "~"+userId.split("~")[1];
+      if (group in stateCopy){
+        let groupState = stateCopy[''+group];
+        resetSequenceStateValues(finalState, groupState);
+        mergeWith(finalState, groupState, mergeCustomizer);
+      }
+    }
+    if (userId in stateCopy){
+      const userState = stateCopy[''+userId];
+      resetSequenceStateValues(finalState, userState);
+      mergeWith(finalState, userState, mergeCustomizer);
+    }
+
+    resetSequenceStateValues(state[''+userId], finalState);
```

Comment:
How is this functionality different than lines 106-108?

Reply:
Here I created a copy of state object so that after merging states (global, group and userId), the original global and group does'nt get affected. So first I am making changes in stateCopy ( merging states and handling conflicts).  Then the final state is mereged in the actual sate object and returning it.